compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -860,6 +860,16 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
 
                 if is_lang_panic_fn(tcx, callee) {
                     self.check_op(ops::Panic);
+
+                    // an internal `panic()` call must have at least one argument
+                    if args[0].ty(&self.ccx.body.local_decls, tcx).kind() != TyKind::Str {
+                        self.tcx.sess.struct_span_err(
```

Comment:
Please add a new struct in the `ops` module (same directory). You can copy paste `struct FnCallIndirect`, which has similar behaviour to what we want here (always hard error, no exceptions, no feature gates). Then you can `self.check_op(ops::YourStruct)` to handle it like we handle all other errors in this file.

Reply:
@oli-obk `ops::Panic` already exists, should I just add this check there?

Reply:
no, `ops::Panic` is about allowing `panic!()` in general, and is a feature gated error. This here is an error about the wrong argument type to `panic!()`, and should for now stay a hard error. Some day we may be more permissive, but it's an entirely independent error (as you can see by the fact that the message you are emitting here (which should stay as the message for the new `ops` struct), is different from the one in `ops::Panic`)